app/models/user.model.js

```diff
@@ -60,6 +59,17 @@ userSchema.pre('save', function(next) {
   });
 });
 
+userSchema.pre('remove', function(next) {
+
+  //remove all salons associated with a user on account delete
+  for (var i in this.salons) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.